### PR TITLE
aws driver updated to collect bucket metadata associated with backend region only

### PR DESCRIPTION
**Pull request by devanshjain7**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
Currently metadata collection is being done for all the buckets irrespective of the region. But now as per the requirement, only buckets associated with the backend region need to be considered for metadata collection. This PR updates the aws driver for fulfilling the requirement.

### CODE REVIEW

File: metadata/pkg/drivers/aws/aws.go [hunk 1]
```diff
@@ -99,43 +98,46 @@ func ObjectList(sess *session.Session, bucket *model.MetaBucket) {
...
 		}
 		buck.BucketTags = tagset
 	}
-
-	ObjectList(sess, buck)
-	if err != nil {
-		return err
-	}
+	bucketArray[buckIdx] = buck
+	*numBuckets += 1
```
Comment (josephjacobmorris) [inline metadata/pkg/drivers/aws/aws.go:140]: may cause race condition

File: metadata/pkg/drivers/aws/aws.go [hunk 2]
```diff
@@ -99,43 +98,46 @@ func ObjectList(sess *session.Session, bucket *model.MetaBucket) {
...
+	if err != nil {
+		return err
 	}
 
 	tags, err := svc.GetBucketTagging(&s3.GetBucketTaggingInput{Bucket: bucket.Name})
 
 	if err != nil && !strings.Contains(err.Error(), "NoSuchTagSet") {
 		log.Errorf("unable to get bucket tags. failed with error: %v", err)
-	} else {
+	} else if err == nil {
```
Comment (josephjacobmorris) [inline metadata/pkg/drivers/aws/aws.go:132]: can refactor to  if err == nil {} else  if !strings.Contains(err.Error(), "NoSuchTagSet")

File: metadata/pkg/drivers/aws/aws.go [hunk 3]
```diff
@@ -99,43 +98,46 @@ func ObjectList(sess *session.Session, bucket *model.MetaBucket) {
 	bucket.NumberOfObjects = numObjects
 	bucket.TotalSize = totSize
 	bucket.Objects = objectArray
+	return nil
 }
 
-func GetBucketMeta(buckIdx int, bucket *s3.Bucket, sess *session.Session, bucketArray []*model.MetaBucket, wg *sync.WaitGroup) error {
+func GetBucketMeta(buckIdx int, bucket *s3.Bucket, sess *session.Session, bucketArray []*model.MetaBucket, numBuckets *int, wg *sync.WaitGroup) error {
```
Comment (josephjacobmorris) [inline metadata/pkg/drivers/aws/aws.go:104]: give meaningful variable names and avoid naming them as i etc

Comment (josephjacobmorris) [inline metadata/pkg/drivers/aws/aws.go:104]: I feel no need for this return value

**Review by josephjacobmorris — CHANGES_REQUESTED** [review]

File: metadata/pkg/drivers/aws/aws.go (hunk 1 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go]: yes thanks for pointing out. its resolved now.

File: metadata/pkg/drivers/aws/aws.go (hunk 3 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go]: done

File: metadata/pkg/drivers/aws/aws.go (hunk 2 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go]: done

File: metadata/pkg/drivers/aws/aws.go (hunk 3 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go]: done

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**